### ISSUE TITLE
Missing semicolon between def yield confusing error

### ISSUE THREAD

**Issue by TristanCacqueray** [byorgey/swarm#79]
**Describe the bug**
This code works:

```
def m1 { move } end
```

But this code:

```
def m1 { move } end
def m2 { m1; m1 } end
```

... fails with `Definitions may only be at the top level: def m1 = move end`

**To Reproduce**
Write the code in a file and load it with `run()`. 

**Expected behavior**
The error should indicate the missing semicolon.
Also when running a file, perhaps the last semicolon shall be required, so that the first program also fails.

**Comment by byorgey** [comment]
Hmm, yes, I think what is happening is that it successfully parses as `(def ... end) (def ... end)`, i.e. the application of one definition to another.  Of course, this makes no sense.   As a first pass, we should definitely put in a special case in the parser to check for this and emit an error suggesting a missing semicolon.   But perhaps there are more general things we could do as well.

**Comment by TristanCacqueray** [comment]
Perhaps the language could support haskell style layout so that it automatically insert a semi-colon on new line if the following line is not indented?

By the way, is there a plan to add tasty validation for such issue?

**Comment by byorgey** [comment]
Parsing for Haskell-style layout is very tricky.  I would not be at all confident in our ability to either get it right or explain it adequately to users.

Hmm, *however*, perhaps we could add one special case: a semicolon can be automatically inserted following an `end` token?

We definitely need some test suites, see #12 .

**Comment by byorgey** [comment]
Closed by #86 .
